org.adempiere.base/src/org/compiere/model/MAttachmentEntry.java

```diff
@@ -71,21 +60,41 @@ public MAttachmentEntry (String name, byte[] data)
 	{
 		this (name, data, 0);
 	}	//	MAttachmentItem
-	
+
+	/**
+	 * Constructor for delayed load
+	 * If you use this constructor, it is required that your storage provider implements the method loadLOBDataEntry for delayed load
+	 * @param parent
+	 * @param name
+	 * @param index
+	 * @param extraObj
+	 */
+	public MAttachmentEntry (MAttachment parent, String name, int index, Object extraObj) {
```

Comment:
Hi @CarlosRuiz-globalqss , changing extraObj to an interface would make this easier to understand and implement. For e.g, ILazyDataSource {public byte[] getData(); public boolean isLoader();} and you can then have FileLazyDataSource implementation that internally holds an File object and lazy load it when getData is call.